java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -0,0 +1,250 @@
+        that they rolled.
+        */
+        return new Captain(
+                name,
+                assignedAbilityScores.get(PHYSICAL),
+                assignedAbilityScores.get(AGILITY),
+                assignedAbilityScores.get(BLASTER),
+                assignedAbilityScores.get(PILOTING),
+                assignedAbilityScores.get(REPAIR),
+                assignedAbilityScores.get(CHARISMA)
+        );
+    }
+
+    private static Map<String, Integer> assignAbilityScores(List<Integer> scoresRolled) {
+        int end = 0;
+        Map<String, Integer> abilityScoreMap = new LinkedHashMap<>();
+
+        while (end == 0) {
+            abilityScoreMap.put(PHYSICAL, null);
+            abilityScoreMap.put(AGILITY, null);
+            abilityScoreMap.put(BLASTER, null);
+            abilityScoreMap.put(PILOTING, null);
+            abilityScoreMap.put(REPAIR, null);
+            abilityScoreMap.put(CHARISMA, null);
+
+            List<Integer> availableScores = new ArrayList<>(scoresRolled);
+            for (Map.Entry<String, Integer> ability : abilityScoreMap.entrySet()) {
+                String abilityName = ability.getKey();
+                System.out.println("");
+                System.out.println("Which score would you like to assign to your [" + abilityName + "] ability?");
+                System.out.println("Enter the number corresponding to the value you wish to use...");
+
+                // TODO: Depending on how Lanterna is set up, we should potentially extract the following code into a
+                //  utility method for players to select a choice from an ordered list
+                DataPrinter.printAsOrderedList(availableScores);
+
+                Integer selectedScore = null;
+                while (selectedScore == null) {
+                    String invalidInputMessage = "Please enter a number from 1 to " + availableScores.size() + ".";
+
+                    try {
+                        selectedScore = Integer.parseInt(scanner.nextLine());
+                    } catch (NumberFormatException e) {
+                        System.out.println(invalidInputMessage);
+                        continue;
+                    }
+                    if (selectedScore < 1 || selectedScore > availableScores.size()) {
+                        System.out.println(invalidInputMessage);
+                        selectedScore = null;
+                    }
+                }
+
+                abilityScoreFlavorText(availableScores.get(selectedScore - 1), abilityName);
+                ability.setValue(availableScores.get(selectedScore - 1));
+                availableScores.remove(selectedScore - 1);
+            }
+
+            printAbilityScoresWithNames(abilityScoreMap);
+            System.out.println("Are these the ability scores you wish to use? (yes/no)");
+            String answer = scanner.nextLine();
```

Comment:
Comment for myself: Add a TODO here for potential yes/no utility method after implementing Lanterna